exporter/awsemfexporter/metadata/metadata.go

```diff
@@ -0,0 +1,124 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metadata
+
+import (
+	"bufio"
+	"io"
+	"os"
+
+	"github.com/aws/aws-sdk-go/aws/ec2metadata"
+	"github.com/aws/aws-sdk-go/aws/session"
+)
+
+const (
+	containerIDLength = 64
+	defaultCGroupPath = "/proc/self/cgroup"
+)
+
+type Metadata interface {
+	GetHostIdentifier() (string, error)
+	GetEC2InstanceID() (string, error)
```

Comment:
I think these are already set into `Resource` by the resource processors. Can we just read the values from `Resource` instead of computing them here again? Exporters should almost never be calculating identifiers I think.

Reply:
I think the host ID inside `Resource` is from SDK side. We need the ID of host where Collector is running. The main purpose of this is to prevent two collectors from using same `logGroup/logStream`. If that conflict happens, the collector cannot send out the logs. In the case if the apps from one host are sending metrics to two collectors, this could happen.

Reply:
I see - let's rename `GetHostIdentifier` to `GetCollectorIdentifier` to make that clearer, and `GetEc2InstanceID` can be private. Though I wonder if it's not simpler to just use a UUID - IIUC, that's how lambda names log streams, based on an execution ID which is effectively a UUID.